server/services/google-cast/lib/google_cast.setValue.js

```diff
@@ -22,6 +23,16 @@ async function setValue(device, deviceFeature, value) {
     client.connect(ipAddress, () => {
       logger.debug('Google Cast Connected, launching app ...');
 
+      if (options.volume) {
+        client.setVolume({ level: options.volume / 100 }, (err, newvol) => {
```

Comment:
You're changing the volume for the voice, but what if the user was playing music, will it be super loud after?

I'm wondering if we could get the previous volume with player.getStatus 
Then rollback the volume after the announce 

What do you think?

Reply:
We can try, I'll check documentation. How does it work when audio notification is terminated on google device? It come back automatically to previous app?

Reply:
I have no idea to be honest! :D

Reply:
Done. But I couldn't do it for Airplay, the lib does not have method to get volume, only set.